### ISSUE TITLE
[Feature] Request Baidu  Kunlun Support in XGrammar Backend

### ISSUE THREAD

**Issue by xyDong0223** [mlc-ai/xgrammar#492]
### Checklist

- [ ] 1. I have searched related issues but cannot get the expected help.
- [ ] 2. The latest version does not have this feature.
- [ ] 3. I have filled all the required information below.
- [ ] 4. The feature request is in English.

### requested feature

# Request for Baidu  Kunlun Support in XGrammar Backend
Hi，
We are the VLLM Kunlun Plugin development team. Currently, we are encountering an issue where `token_bitmask_inplace_kernel` is not supported when using xgrammar. We hope to add kernel support for this. Thank you very much for your reply and communication！

# Pytoch implement
Could add  option： 
`@torch.compile(dynamic=True)`
`def apply_token_bitmask_eager(logits, bitmask, vocab_size):`
to
`@torch._dynamo.disable`

Like
`
_DISABLE_TOKEN_BITMASK_COMPILE = (
    os.getenv("VLLM_DISABLE_TOKEN_BITMASK_COMPILE", "0") == "1"
)`
`_apply_token_bitmask_eager = torch._dynamo.disable(
    _apply_token_bitmask_eager
)`

# Kernel implement
If possible, we would be very happy to collaborate on the kernel implementation！！
 

### Motivation

# Motivation
With VLLM and function call becoming mainstream choices for AI inference, we are eager to bring Xgrammar functionality to Kunlun XPU users!

**Comment by Ubospica** [comment]
Contribution to Kunlun kernel is welcome!

About torch.compile, could you establish the reason for the crash? Why enabling torch.compile in both vllm and xgrammar leads to a crash?

**Comment by Ubospica** [comment]
Closed by #494. Feel free to reopen if there are further problems!
